src/stores/deck.js

```diff
@@ -36,9 +36,12 @@ const useDeckStore = create((set, get) => ({
     set(state => ({
       deck: shuffleDeck(state.deck),
     })),
-  resetDeck: () =>
-    set(() => ({
-      deck: shuffleDeck(generateDeck()),
+  resetDeck: () => {
+    set({ deck: shuffleDeck(generateDeck()) });
+  },
+  addCardToDeck: card =>
+    set(state => ({
+      deck: [...state.deck, card],
```

Comment:
How about this calls shuffle on the array? That way the new card is immediately shuffled in. So just wrap what you have: 
```
deck: shuffleDeck([...state.deck, card]),
```